test/e2e/test_image.py

```diff
@@ -95,3 +96,177 @@ def test_create_with_invalid_image_file(self):
 
         self.post_image(IMAGE_POST_METADATA_DATA_REQUIRED_VALUES_ONLY, "invalid_image.jpg")
         self.check_post_image_failed_with_detail(422, "File given is not a valid image")
+
+
+# pylint:disable=fixme
+# TODO: Inherit from GetDSL when added
+class ListDSL(CreateDSL):
+    """Base class for list tests."""
+
+    _get_response_image: Response
+
+    def get_images(self, filters: Optional[dict] = None) -> None:
+        """
+        Gets a list of images with the given filters.
+
+        :param filters: Filters to use in the request.
+        """
+        self._get_response_image = self.test_client.get("/images", params=filters)
+
+    def check_get_images_success(self, expected_images_get_data: list[dict]) -> None:
+        """
+        Checks that a prior call to `get_images` gave a successful response with the expected data returned.
+
+        :param expected_images_get_data: List of dictionaries containing the expected image data as would
+            be required for an `ImageSchema`.
+        """
+        assert self._get_response_image.status_code == 200
+        assert self._get_response_image.json() == expected_images_get_data
+
+    def check_images_list_response_failed_with_message(self, status_code, expected_detail, obtained_detail):
+        """Checks the response of listing images failed as expected"""
+
+        assert self._get_response_image.status_code == status_code
+        assert obtained_detail == expected_detail
```

Comment:
Could these go below the post_test_images so the order is consistent with IMS please.